### PR TITLE
Added test coverage

### PR DESCRIPTION
**Pull request by 20k-ultra**

### CODE REVIEW

File: src/versionbot-utils.ts [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
 
 const sleep = (milliseconds: number) => {
 	return new Promise((resolve) => setTimeout(resolve, milliseconds));
 };
 
-export async function getBranch(pr: number): Promise<string> {
+export async function getBranch(context: RepoContext): Promise<string> {
+	if (!context.pullRequest) {
+		throw new Error(
+			'Cannot find Versionbot branch for non-PR context: Json.stringify(context)',
```
Comment (klutchell) [inline src/versionbot-utils.ts:15]:
```suggestion
			`Cannot find Versionbot branch for non-PR context: Json.stringify(context)`,
```

File: tests/lib/sleep.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default async function sleep(ms: number): Promise<void> {
+	return new Promise((resolve) => setTimeout(resolve, ms));
+}
```
Comment (Page-) [inline tests/lib/sleep.ts:3]:
```suggestion
export const sleep = util.promisify(setTimeout);
}
```
btw `import { setTimeout } from 'timers/promises'` would be better but it's not available in the nodejs 12 used here (but updating nodejs may well make sense)

File: tests/src/main.spec.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+			'balenaTokenExample',
+		);
+
+		while (!actionStub.lastCall) {
+			// Since the main file performs async operations without a way to know
+			// once it is done we will wait until the action is called.
+			// The test timeout will make this fail if it runs for longer then allowed time.
+			await new Promise((resolve) => {
+				setTimeout(resolve, 1);
+			});
```
Comment (Page-) [inline tests/src/main.spec.ts:85]:
```suggestion
			await sleep(1);
```

File: tests/src/balena-utils.spec.ts [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+			).to.eventually.equal(149241);
+		});
+	});
+
+	describe('init', () => {
+		let sdkStub: SinonStub;
+
+		before(() => {
+			sdkStub = stub(balena, 'getSdk');
+			sdkStub = sdkStub.returns({
```
Comment (20k-ultra) [inline tests/src/balena-utils.spec.ts:172]: I don't have to redelcare the sdkStub just call returns function in it..

File: tests/src/action.spec.ts [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+		pushStub.restore();
+		vbBranchStub.restore();
+		fetchStub.restore();
+		checkoutStub.restore();
+		createTagStub.restore();
+		releaseVersionStub.restore();
+		releaseByTagStub.restore();
+		setOutputStub.restore();
+		errorLogStub.restore();
+	});
```
Comment (20k-ultra) [inline tests/src/action.spec.ts:77]: maybe put these into an array and for loop over them to reset/store

**Comment by 20k-ultra** [comment]
@balena-ci I self-certify!

**Pull request merged**